src/github-handler/comment-handler/remote-patch-ranges-handler.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Octokit} from '@octokit/rest';
+import {FileRanges, PatchText, Range, RepoDomain} from '../../types';
+import {getGitHubPatchRanges} from './github-patch-format-handler';
+import {logger} from '../../logger';
+
+/**
+ * For a pull request, get each remote file's patch text asynchronously
+ * Also get the list of files whose patch data could not be returned
+ * @param {Octokit} octokit the authenticated octokit instance
+ * @param {RepoDomain} remote the remote repository domain information
+ * @param {number} pullNumber the pull request number
+ * @param {number} pageSize the number of results to return per page
+ * @returns {Promise<Object<PatchText, string[]>>} the stringified patch data for each file and the list of files whose patch data could not be resolved
+ */
+export async function getCurrentPulLRequestPatches(
+  octokit: Octokit,
+  remote: RepoDomain,
+  pullNumber: number,
+  pageSize: number
+): Promise<{patches: PatchText; filesMissingPatch: string[]}> {
+  // TODO support pagination
+  const filesMissingPatch: string[] = [];
+  const files = (
+    await octokit.pulls.listFiles({
+      owner: remote.owner,
+      repo: remote.repo,
+      pull_number: pullNumber,
+      per_page: pageSize,
+    })
+  ).data;
+  const patches: PatchText = new Map<string, string>();
+  if (files.length === 0) {
+    logger.error(
+      `0 file results have returned from list files query for Pull Request #${pullNumber}. Cannot make suggestions on an empty Pull Request`
+    );
+    throw Error('Empty Pull Request');
+  }
+  files.forEach(file => {
+    if (file.patch === undefined) {
```

Comment:
Might this ever be null or empty string?

Reply:
It shouldn't, but if that's the case, then we assume that getting the patch data was successful. We leave it to `patchTextToRanges` to throw an error since the get was not erroneous, but the parsing. We should never expect null, nor an empty string, nor a string without any hunk text.